Lib/test/test_typing.py

```diff
@@ -3654,6 +3705,22 @@ def __init__(self):
 
         Foo()  # Previously triggered RecursionError
 
+    def test_interaction_with_isinstance_checks_on_superclasses_with_ABCMeta(self):
```

Comment:
This could fail spuriously if previous code happened to do `isinstance(1, Sized)` before this test, right? Would be good if we could explicitly clear the ABC cache here.

Reply:
Done in 0b13ae0!

Reply:
Had to use a private method, but the docstring does explicitly say that the method is for "debugging or testing", and that's what we're doing here:

https://github.com/python/cpython/blob/df396b59af9d50892e5e30463300e8458cb84263/Lib/abc.py#L137-L139C33